lib/winapi/findfile.c

```diff
@@ -0,0 +1,129 @@
+#include <string.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <winbase.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <hal/fileio.h>
+#include <hal/winerror.h>
+#include <fileapi.h>
+
+struct FileInfo {
+    FILE_DIRECTORY_INFORMATION dirInfo;
+    char filename[MAX_PATH];
+};
+
+static void dirtofind (FILE_DIRECTORY_INFORMATION *dirInfo, LPWIN32_FIND_DATAA lpFindFileData)
+{
+    lpFindFileData->dwFileAttributes = dirInfo->FileAttributes;
+    lpFindFileData->ftCreationTime.dwLowDateTime = dirInfo->CreationTime.LowPart;
+    lpFindFileData->ftCreationTime.dwHighDateTime = dirInfo->CreationTime.HighPart;
+    lpFindFileData->ftLastAccessTime.dwLowDateTime  = dirInfo->LastAccessTime.LowPart;
+    lpFindFileData->ftLastAccessTime.dwHighDateTime = dirInfo->LastAccessTime.HighPart;
+    lpFindFileData->ftLastWriteTime.dwLowDateTime  = dirInfo->LastWriteTime.LowPart;
+    lpFindFileData->ftLastWriteTime.dwHighDateTime = dirInfo->LastWriteTime.HighPart;
+    lpFindFileData->nFileSizeHigh = dirInfo->EndOfFile.HighPart;
+    lpFindFileData->nFileSizeLow = dirInfo->EndOfFile.LowPart;
+    memcpy(lpFindFileData->cFileName, dirInfo->FileName, dirInfo->FileNameLength);
+    lpFindFileData->cFileName[dirInfo->FileNameLength] = '\0';
+    lpFindFileData->cAlternateFileName[0] = '\0';
```

Comment:
Should this `cAlternateFileName` field be populated with the 8.3 name for compatibility?

Reply:
No, not done in XAPI either.